dashboard/app/models/sections/section.rb

```diff
@@ -539,57 +517,100 @@ def summarize(include_students: true)
+          login_type_name: login_type_name,
+          participant_type: participant_type,
+          course_display_name: course_display_name,
+          course_offering_id: course_offering_id,
+          course_version_id: unit_group ? unit_group&.course_version&.id : script&.course_version&.id,
+          unit_id: unit_group ? script_id : nil,
+          unitPosition: unit_group_unit&.position,
+          course_id: course_id,
+          script: {
+            id: selected_unit&.id,
+            name: selected_unit&.name,
+            project_sharing: selected_unit&.project_sharing
+          },
+          studentCount: num_students,
+          providerManaged: provider_managed?,
+          hidden: hidden,
+          students: include_students ? unique_students.map(&:summarize) : nil,
+          restrict_section: restrict_section,
+          is_assigned_csa: assigned_csa?,
+          is_assigned_single_unit_course: unit_group&.single_unit_course?,
+          # this will be true when we are in emergency mode, for the scripts returned by ScriptConfig.hoc_scripts and ScriptConfig.csf_scripts
+          post_milestone_disabled: !!script && !Gatekeeper.allows('postMilestone', where: {script_name: script.name}, default: true),
+          code_review_expires_at: code_review_expires_at,
+          sync_enabled: Policies::Lti.roster_sync_enabled?(teacher),
+          ai_tutor_enabled: ai_tutor_enabled,
+          at_risk_age_gated_date: at_risk_student&.at_risk_age_gated_date,
+          at_risk_age_gated_us_state: at_risk_student&.us_state,
+          avatar_color: avatar_color,
+          avatar_emoji: avatar_emoji,
+        }
+      )
+    end
+  end
+
+  # A very abridged version of summarize that shows only information needed by participants.
+  def summarize_for_participant
+    ActiveRecord::Base.connected_to(role: :reading) do
+      base_url = CDO.studio_url('/teacher_dashboard/sections/')
+
+      title = ''
+      link_to_assigned = base_url
+      title_of_current_unit = ''
+      link_to_current_unit = ''
+
+      if unit_group
+        title = unit_group.localized_title
+        link_to_assigned = course_path(unit_group)
+        if script_id
+          title_of_current_unit = script.title_for_display
+          link_to_current_unit = if Policies::Courses.modularity_enabled? && unit_group_unit
+                                   course_unit_path(unit_group, unit_group_unit.position)
+                                 else
+                                   script_path(script)
+                                 end
+        end
+      elsif script_id
+        title = script.title_for_display
+        link_to_assigned =
+          if unit_group_unit
+            if Policies::Courses.modularity_enabled?
+              course_unit_path(unit_group_unit.unit_group, unit_group_unit.position)
+            else
+              script_path(script)
+            end
+          else
+            script_path(script)
+          end
```

Comment:
this looks like it can now be simplified
```suggestion
        link_to_assigned =
          if Policies::Courses.modularity_enabled?(course_unit_path(unit_group_unit.unit_group, unit_group_unit.position)) && unit_group_unit
          else
            script_path(script)
          end
```